app/models/veteran.rb

```diff
@@ -178,7 +178,7 @@ def incident_flash?
 
   # Postal code might be stored in address line 3 for international addresses
   def zip_code
-    zip_code = bgs_record&.[](:zip_code)
+    zip_code = bgs_record_found? ? bgs_record&.[](:zip_code) : ""
```

Comment:
Can you add a test, if one doesn't already exists, that handles this scenario.

Reply:
Wondering if you should just `return ""` so that it's clear the rest of the method is irrelevant?
```suggestion
    zip_code = bgs_record_found? ? bgs_record&.[](:zip_code) : return ""
```
Another question: do other callers of `zip_code` handle empty string appropriately?

Reply:
Ah, good point. I just added and pushed a test for this.

Reply:
Would pulling that return out into a guard clause like this make sense?
```
  def zip_code
    return "" if !bgs_record_found? 

    zip_code = bgs_record&.[](:zip_code)
    zip_code ||= (@address_line3 if (@address_line3 || "").match?(Address::ZIP_CODE_REGEX))
```

Reply:
Yup! Even better!

Reply:
I'm not sure about the callers of `zip_code`. I was making the assumption that an empty string would be better than an error, I'll see what I can find.

Reply:
Have any commentary on this?
> Another question: do other callers of zip_code handle empty string appropriately?

Reply:
Sometimes returning `nil` is the better option. For example, `nil` could represent it being not available from BGS, whereas an empty string may mean that BGS has an empty string saved in the zip code field, which may indicate some other interpretation. Whatever the decision, just ensure that callers of `zip_code` are aware of the possibility to return empty string -- they may want to react differently if `nil` is returned. 🤷

Reply:
Thanks! That's really helpful. I misunderstood what `zip_code` was returning. I thought it used to always return an empty string (nope).

Looking through the `veterans_spec.rb` file there are other tests that explicitly expect zip_code to be nil. Changing this to return nil is much more consistent with what it was doing AND what the tests expect. Thanks!

Reply:
Please also check callers of `zip_code` in our application code (outside our tests) to see how they are reacting to `nil` versus some other value, which may or may not be a valid zip code. A quick grep shows that both Caseflow Hearings and Intake use `zip_code`; I recommend posting this PR to #appeals-engineering to get more 👀 on this.

Reply:
Putting my notes here before posting to #appeals-engineering as Yoom suggested.

I looked through the ruby/rails code and it looks to me like we already expect zip to be nil in address.rb. I don't see any obvious effects in the other three files that seemed likely candidates: veteran.rb, appeal.rb, legacy_appeal.rb. Long notes are below.


I looked through the javascript code and as far as I can tell (long form notes below), zip_code is displayed only in a way that's not sensitive to the difference between an empty string, like this:
`<span>{city}, {state} {zip} {country === 'USA' ? '' : country}</span>`
The one exception is AddressForm.jsx, which is only used by AddClaimantPage.jsx, which is a foxtrot WIP.

**Backend Notes (from searching 'zip' in files with names containing 'rb'):**
```
# Uses zip
veteran.rb - This is the file I changed, I don't see any adverse effects.
appeal.rb - Delegates zip to veteran.rb, no obvious effect from this change
legacy_appeal.rb - Delegates zip to veteran.rb, no obvious effect from this change
address.rb - Deals with zips, already expects zip to be nil (rather than "")
# Never checks the value of zip
veteran_serializer.rb - Renames zip, doesn't care about the value stored.
veteran_details_serializer.rb - Renames zip, doesn't check the value.
address_mapper.rb - Renames zip, doesn't check the value.
```

**Frontend Notes (from searching 'zip' in files with names containing 'js'):**
```
// Display only
IssueRemandReasonsOptions.jsx - Using lodash's _.zip
AssignHearingModal.jsx - Displays zip as string
Address.jsx - Displays zip as string
utils.js - Renames variable, doesn't change zip
constants.js - Using lodash's _.zip
ErrorAlert.jsx - Reacts to zip, not sensitive to this change.
DailyDocketRowDisplayText.jsx - Displays zip as string
AssignHearingsFields.jsx - Displays zip as string
// Display only, uses Address.jsx
RepresentativeSection.jsx - Passes zip to Address.jsx, display only
AppellantSection.jsx - Passes zip to Address.jsx, display only
ScheduleVeteran.jsx - Sends zip to ScheduleVeteranForm.jsx
ScheduleVeteranForm.jsx - Passes it to Address.jsx
Address.jsx - Displays zip as string (there are two versions of this file, they display zip in the same way)
// Using location zip, not veteran
AppealHearingLocations.jsx - Not using the veteran zip code, using location zip code.
// Allows editing
AddressForm.jsx - Only used by the AddClaimantPage, as far as I can tell this form is non-functional (Continue never lights up).
```